pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -1157,98 +1170,17 @@ class ModelParser implements Parser {
         return null;
     }
 
-    /**
-     * Works like a regular Java cast, except if the value doesn't match the type, return null
-     * instead of throwing an exception.
-     */
-    protected <X> X castOrNull(Class<X> type, Object value) {
-        if (type.isInstance(value))
-            return type.cast(value);
-        return null;
-    }
+    public static final boolean AST_DEBUG_LOGGING = SystemProperties.getBoolean(ModelParser.class.getName()+".astDebugLogging")
```

Comment:
http://javadoc.jenkins.io/hudson/Main.html#isDevelopmentMode

But is has been iffy for me in the past, so I think I looked for some maven thing in `System.getProperty()` or something. I'll see what I can find.